TrashBin.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "Cat.h"
+#include "Fruit.h"
+#include "LeakedObjectDetector.h"
+#include "Treadmill.h"
+
+struct TrashBin
+{
+    TrashBin (Cat& nearbyCat);
```

Comment:
Why does this `Cat` need to be a reference and not a `const &`? 
The code in the `TrashBin` constructor does not modify the instance.
- [ ] investigate and revise if possible.

Reply:
It can be const.  When I read "function implementations" I did not mentally include the constructors, for some reason.  I've gone back through and updated all constructors.